src/modules/escrow/services/escrow.transaction.service.ts

```diff
@@ -1,41 +1,48 @@
-import { Address, AddressValue, TokenTransfer } from '@multiversx/sdk-core/out';
+import {
+    Address,
+    AddressValue,
+    Token,
+    TokenTransfer,
+} from '@multiversx/sdk-core';
 import { Injectable } from '@nestjs/common';
-import BigNumber from 'bignumber.js';
-import { gasConfig, mxConfig } from 'src/config';
+import { gasConfig } from 'src/config';
 import { InputTokenModel } from 'src/models/inputToken.model';
 import { TransactionModel } from 'src/models/transaction.model';
+import { TransactionOptions } from 'src/modules/common/transaction.options';
 import { MXProxyService } from 'src/services/multiversx-communication/mx.proxy.service';
 
 @Injectable()
 export class EscrowTransactionService {
     constructor(private readonly mxProxy: MXProxyService) {}
 
     async withdraw(senderAddress: string): Promise<TransactionModel> {
-        const contract = await this.mxProxy.getEscrowContract();
-
-        return contract.methodsExplicit
-            .withdraw([new AddressValue(Address.fromString(senderAddress))])
-            .withChainID(mxConfig.chainID)
-            .withGasLimit(gasConfig.escrow.withdraw)
-            .buildTransaction()
-            .toPlainObject();
+        return await this.mxProxy.getEscrowSmartContractTransaction(
+            new TransactionOptions({
+                sender: senderAddress,
```

Comment:
This `senderAddress` is not correct. You still need to have a `userAddress` argument for this endpoint and use it as sender. The `senderAddress` refers to the address who sent funds to `userAddress`

Reply:
done